client/src/templates/Challenges/redux/execute-challenge-saga.js

```diff
@@ -78,19 +77,16 @@ export function* executeCancellableChallengeSaga(payload) {
     }
   }
 
-  if (previewTask) {
-    yield cancel(previewTask);
-  }
   // executeChallenge with payload containing {showCompletionModal}
   const task = yield fork(executeChallengeSaga, payload);
-  previewTask = yield fork(previewChallengeSaga, { flushLogs: false });
+  yield fork(previewChallengeSaga, { flushLogs: false });
 
   yield take(actionTypes.cancelTests);
   yield cancel(task);
 }
 
 export function* executeCancellablePreviewSaga() {
-  previewTask = yield fork(previewChallengeSaga);
+  yield fork(previewChallengeSaga);
 }
```

Comment:
Since we never try to cancel a preview saga, we don't need this generator function. Instead, could you replace `executeCancellablePreviewSaga` in `createExecuteChallengeSaga` with `previewChallengeSaga`?

Reply:
Updated the PR.